bin/download-spark.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+bin=$(dirname "${BASH_SOURCE-$0}")
+bin=$(cd "${bin}">/dev/null; pwd)
+
+. "${bin}/common.sh"
+
+ZEPPELIN_ENV="conf/zeppelin-env.sh"
+ZEPPELIN_ENV_TEMP="${ZEPPELIN_ENV}.template"
+ZEPPELIN_VERSION="$(getZeppelinVersion)"
+
+SPARK_VERSION="2.0.0"
+HADOOP_VERSION="2.7"
+
+SPARK_CACHE="local-spark"
+SPARK_ARCHIVE="spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}"
+ANSWER_FILE="README.txt"
+
+# Download Spark binary package from the given URL.
+# Ties 3 times with 1s delay
+# Arguments: url - source URL
+
+trap "download_with_retry; rm -r ${ZEPPELIN_HOME}/${SPARK_CACHE}; exit 1" SIGTERM SIGINT SIGQUIT
+
+function download_with_retry() {
+  local url="$1"
+  curl -O --retry 3 --retry-delay 1 "${url}"
+
+  if [[ "$?" -ne 0 ]]; then
+    echo -e "\nStop downloading with unexpected error.\nPlease restart Zeppelin if you want to download local Spark again."
+  fi
+}
+
+function unzip_spark_bin() {
+  if ! tar zxf "${SPARK_ARCHIVE}.tgz" ; then
+    echo "Unable to extract ${SPARK_ARCHIVE}.tgz" >&2
+    rm -rf "${SPARK_ARCHIVE}"
+  else
+    set_spark_home
+  fi
+
+  rm -f "${SPARK_ARCHIVE}.tgz"
+}
+
+function check_zeppelin_env() {
+  if [[ ! -f "${ZEPPELIN_ENV}" ]]; then
+    echo -e "\n${ZEPPELIN_ENV} doesn't exist\nCreating ${ZEPPELIN_ENV} from ${ZEPPELIN_ENV_TEMP}..."
+    cp "${ZEPPELIN_HOME}/${ZEPPELIN_ENV_TEMP}" "${ZEPPELIN_HOME}/${ZEPPELIN_ENV}"
+  fi
+}
+
+function set_spark_home() {
+  local line_num
+  check_zeppelin_env
```

Comment:
After `check_zeppelin_env`, if there is already `conf/zeppelin-env.sh` exists, which means user explicitly created it, then i think we shouldn't change it.